Modules/Loadable/Markups/MRMLDM/vtkMRMLMarkupsDisplayableManager.cxx

```diff
@@ -138,6 +148,21 @@ void vtkMRMLMarkupsDisplayableManager::PrintSelf(ostream& os, vtkIndent indent)
     }
 }
 
+//---------------------------------------------------------------------------
+void vtkMRMLMarkupsDisplayableManager::RegisterMarkup(vtkMRMLMarkupsNode* node,
+                                                      vtkSlicerMarkupsWidget* widget,
+                                                      const std::string& name)
+{
+  if (!node)
+    {
+    vtkErrorMacro("RegisterMarkup: invalid node.");
+    }
+
+  this->MarkupsNodesWidgets[node->GetClassName()] = widget;
```

Comment:
We may have multiple widgets for the same node class. For example, all Python scripted widgets will use the same vtkSlicerMarkupsScriptedWidget class. We may also add different widgets for the same markups node (for example, we may add slightly different widgets or widget representations for the same vtkMRMLMarkupsPlaneNode).

As a consequence, we cannot use the node class name as key in the MarkupsNodesWidgets map. Instead, we could specify a "Type" attribute in the widget class, which would return a string (e.g., "Points", "Plane", "ClosedCurve", "CurvedPlane", ...).

To make sure that the design is usable in Python, it would be important to implement a Python scripted widget (it can be a test in Slicer core or in your extension).